jgrapht-core/src/test/java/org/jgrapht/alg/scoring/AlphaCentralityTest.java

```diff
@@ -0,0 +1,288 @@
+        assertEquals(pr.getVertexScore("c"), 5.2500, 0.0001);
+    }
+
+    public void testweightedGraph2()
+    {
+        DirectedWeightedPseudograph<String, DefaultWeightedEdge> g =
+            new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+
+        g.addVertex("center");
+        g.addVertex("a");
+        g.addVertex("b");
+        g.addVertex("c");
+
+        g.setEdgeWeight(g.addEdge("center", "a"), 1.0);
+        g.setEdgeWeight(g.addEdge("center", "b"), 1.0);
+        g.setEdgeWeight(g.addEdge("center", "c"), 1.0);
+
+        VertexScoringAlgorithm<String, Double> pr = new AlphaCentrality<>(g, 0.85, 1.0, 100, 0.0001);
+
+        assertEquals(pr.getVertexScore("center"), 1.0000, 0.0001);
+        assertEquals(pr.getVertexScore("a"), 1.8500, 0.0001);
+        assertEquals(pr.getVertexScore("b"), 1.8500, 0.0001);
+        assertEquals(pr.getVertexScore("c"), 1.8500, 0.0001);
+    }
+
+    public void testUnweightedGraph2()
+    {
+        DirectedPseudograph<String, DefaultEdge> g = new DirectedPseudograph<>(DefaultEdge.class);
+
+        g.addVertex("center");
+        g.addVertex("a");
+        g.addVertex("b");
+        g.addVertex("c");
+        g.addVertex("d");
+
+        g.addEdge("center", "a");
+        g.addEdge("center", "b");
+        g.addEdge("center", "c");
+
+        VertexScoringAlgorithm<String, Double> pr = new AlphaCentrality<>(g, 0.85, 1.0, 100, 0.0001);
+
+        assertEquals(pr.getVertexScore("center"), 1.0000, 0.0001);
+        assertEquals(pr.getVertexScore("a"), 1.8500, 0.0001);
+        assertEquals(pr.getVertexScore("b"), 1.8500, 0.0001);
+        assertEquals(pr.getVertexScore("c"), 1.8500, 0.0001);
+        assertEquals(pr.getVertexScore("d"), 1.0000, 0.0001);
+    }
+
+    public void testEmptyGraph()
+    {
+        DirectedPseudograph<String, DefaultEdge> g = new DirectedPseudograph<>(DefaultEdge.class);
+
+        VertexScoringAlgorithm<String, Double> pr = new AlphaCentrality<>(g, 0.85, 1.0, 100, 0.0001);
+
+        assertTrue(pr.getScores().isEmpty());
+    }
+
+    public void testNonExistantVertex()
+    {
+        DirectedPseudograph<String, DefaultEdge> g = new DirectedPseudograph<>(DefaultEdge.class);
+
+        g.addVertex("center");
+        g.addVertex("a");
+        g.addVertex("b");
+        g.addVertex("c");
+        g.addVertex("d");
+
+        g.addEdge("center", "a");
+        g.addEdge("center", "b");
+        g.addEdge("center", "c");
+
+        VertexScoringAlgorithm<String, Double> pr = new AlphaCentrality<>(g, 0.85, 1.0, 100, 0.0001);
+
+        try {
+            pr.getVertexScore("unknown");
+            fail("No!");
+        } catch (IllegalArgumentException e) {
+        }
+    }
+
+    public void testBadParameters()
+    {
+        DirectedPseudograph<String, DefaultEdge> g = new DirectedPseudograph<>(DefaultEdge.class);
+
+        try {
+            new AlphaCentrality<>(g, 1.1, 1.0, 100, 0.0001);
+            fail("No!");
```

Comment:
Better would be to use the JUnit4 annotation `@Test(expected=IllegalArgumentException)`